src/helpers/namespace.ts

```diff
@@ -46,5 +46,124 @@ export type CreateManyInput<T> =
       set?: T | T[];
     };
 
+/**
+ * A typed version of NestedStringFilter, allowing narrowing of string types to discriminated unions.
+ */
+export type TypedNestedStringFilter<T extends string> = Prisma.StringFilter & {
```

Comment:
Would you mind changing `T` to `S` just because it can be only an string? 🤡

Reply:
Haha, yes definitely!